### PR TITLE
Display the resource file doesn't exist message in task create page

### LINKED ISSUE #15349: [Bug] [Workflow Definition] [UI] Non-existent resource front ends cannot be culled
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

After a resource is deleted or its name is changed, the resource referenced by the current task node cannot be displayed or deleted. As a result, a message indicating that the resource does not exist is displayed during task running

### What you expected to happen

It should be possible to remove non-existent resource options

### How to reproduce

Select the resource, save the workflow, and go to the resource center to change the reference resource name

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ly109974**
fix #15349 

https://github.com/apache/dolphinscheduler/assets/103415610/8f4aa762-dd30-4dd8-9a1e-d9ba52f8b3fc

### CODE REVIEW

File: dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-resources.ts [hunk 1]
```diff
@@ -33,6 +34,7 @@ export function useResources(
   const resourcesLoading = ref(false)
 
   const taskStore = useTaskNodeStore()
+  const allResourceFullName = ref([] as string[])
```
Comment (songjianet) [inline dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-resources.ts]: ref<Array<string>>([]) or ref<string[]>([])

**Review by songjianet — CHANGES_REQUESTED** [review]

File: dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-resources.ts [hunk 2]
```diff
@@ -69,12 +101,36 @@ export function useResources(
...
+          value.forEach((item) => {
+            if (!validateResourceExist(item)) {
+              errorNames.push(item)
+            }
+          })
+          if (errorNames.length > 0) {
+            errorNames.forEach((item) => {
+              // delete select node
+              value.splice(value.indexOf(item), 1)
+              resourcesOptions.value.push({
```
Comment (songjianet) [inline dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-resources.ts]: The type assigned to resourcesOptions does not match the definition.

**Review by songjianet — CHANGES_REQUESTED** [review]

**Comment by ly109974** [comment]
image
image
optimize adjusted

**Comment by ly109974** [comment]
@songjianet  Please take a look

**Review by songjianet — APPROVED** [review]

**Pull request merged**